app/code/Magento/Catalog/Model/Product.php

```diff
@@ -2084,14 +2085,11 @@ public function getCustomOptions()
      * Get product custom option info
      *
      * @param   string $code
-     * @return  array
+     * @return  OptionInterface|null
```

Comment:
Theoretically, this class can be used in WebAPI requests.

The current implementation of WebAPI "request/response" parser has a limitation:

Valid object types include a fully qualified class name or a fully qualified interface name.
https://devdocs.magento.com/guides/v2.3/extension-dev-guide/service-contracts/service-to-web-service.html

It's OK to change dockblock but please replace short namespace on "fully qualified name" in methods dockblocks
Thanks